src/main/java/uk/gov/hmcts/darts/authorisation/config/AuthorisationAspect.java

```diff
@@ -0,0 +1,259 @@
+
+    private static final String HEARINGS_PATH = "hearings";
+    private static final String HEARING_ID_PARAM = "hearing_id";
+
+    private static final String AUDIO_REQUESTS_PATH = "audio-requests";
+    private static final String AUDIO_REQUEST_ID_PARAM = "media_request_id";
+
+    private static final String AUDIOS_PATH = "audio/preview"; //FIXME should probably be "audios"
+    private static final String AUDIO_ID_PARAM = "media_id";
+
+    private static final String TRANSCRIPTIONS_PATH = "transcriptions";
+    private static final String TRANSCRIPTION_ID_PARAM = "transcription_id";
+
+    private static final Pattern CASES_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, CASES_PATH));
+
+    private static final Pattern HEARINGS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, HEARINGS_PATH));
+
+    private static final Pattern AUDIO_REQUESTS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, AUDIO_REQUESTS_PATH));
+
+    private static final Pattern AUDIOS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, AUDIOS_PATH));
+
+    private static final Pattern TRANSCRIPTIONS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, TRANSCRIPTIONS_PATH));
+
+    private final Authorisation authorisation;
+
+    @Before("@annotation(uk.gov.hmcts.darts.authorisation.annotation.Authorisation)")
+    public void authorisation(JoinPoint joinPoint) throws Throwable {
+        HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+
+        uk.gov.hmcts.darts.authorisation.annotation.Authorisation authorisationAnnotation = ((MethodSignature) joinPoint.getSignature()).getMethod()
+            .getAnnotation(uk.gov.hmcts.darts.authorisation.annotation.Authorisation.class);
+
+        Set<SecurityRoleEnum> roles = Set.of(authorisationAnnotation.securityRoles());
+        if (roles.isEmpty()) {
+            throw new DartsApiException(AuthorisationError.USER_NOT_AUTHORISED_FOR_COURTHOUSE);
+        }
+
+        ContextIdEnum contextId = authorisationAnnotation.contextId();
+
+        switch (contextId) {
+            case CASE_ID -> checkAuthorisationByCaseId(request, roles);
+            case HEARING_ID -> checkAuthorisationByHearingId(request, roles);
+            case MEDIA_REQUEST_ID -> checkAuthorisationByMediaRequestId(request, roles);
+            case MEDIA_ID -> checkAuthorisationByMediaId(request, roles);
+            case TRANSCRIPTION_ID -> checkAuthorisationByTranscriptionId(request, roles);
+            default -> log.warn("Unrecognised contextId");
+        }
+    }
+
+    private void checkAuthorisationByCaseId(HttpServletRequest request, Set<SecurityRoleEnum> roles) {
+        Optional<String> caseIdParamOptional = Optional.empty();
+
+        Matcher matcher = CASES_ID_PATH_PATTERN.matcher(request.getRequestURI());
```

Comment:
Rather than binding ourselves to the URL pattern can we not just check the input parameters of the method? Spring has already extrapolated these for us.

Reply:
Thanks @mestebanez - done in commit 023357fd. I missed that, now simplified and clears my headache around the regex matching of our audio endpoints such as "preview". Cheers